### DISCUSSION TITLE
MAXIMUN IMAGE GENERATION ALLOWED = 50

### DISCUSSION THREAD

**Discussion by LIQUIDMIND111** [hlky/stable-diffusion-webui#1096] [Q&A]
HOW TO SET THE MAXIMUM image generation from 50 BACK to 250 as OLDER VERSIONS? Older HLKY interfaces had the ability to allow up to 250 images of generation per session. NOW its only up yo 50. Since i want to make ANIMATIONS with seeds variants, i need MORE frames so i can get more animation time, and OLDER versions can allow UP TO 250 images generation when you clicked PROCESS. Now, this models allow only up to 50, so 50 can make a good 2-second animation at 25fps. CAN ANYONE HELP ME TO CHANGE THE CODE OR SOMETHING
![Screenshot 2022-09-13 161020](https://user-images.githubusercontent.com/112510084/189999649-c1d80421-5b22-4748-b705-6aed9ee63bbb.jpg)
???

**Comment by codedealer** [comment]
Go to `frontend/frontend.py` in your local installation and find this line https://github.com/sd-webui/stable-diffusion-webui/blob/e6a9e5d968aad020e54045a0923d8c6aa0b54591/frontend/frontend.py#L39
Change `maximum` to whatever you want.

**Comment by PierreHoule** [comment]
Another solution is to type "-n 100" without quotes (or any other number larger than 50) in the prompt field. As soon as you hit the spacebar after typing this, the "Number of images to generate" slider moves up to the specified number.

**Reply by codedealer** [reply]
This no longer works, prompt parsing was disabled because of negative impact on performance.

**Comment by LIQUIDMIND111** [comment]
Nice. Thanks so much!

**Comment by LIQUIDMIND111** [comment]
No problem I already found a solution using Google inspect or just changing
the code in the frontend file

**Comment by Wonderflex** [comment]
If you want to be able to type in any number you like, then change

 https://github.com/sd-webui/stable-diffusion-webui/blob/e6a9e5d968aad020e54045a0923d8c6aa0b54591/frontend/frontend.py#L39

to

`txt2img_batch_count = gr.Number(label='Number of images to generate', precision=0,`

This will give you a text box to enter any number you like, with a little up/down arrow to progress by one if you prefer clicking.

**Reply by dmt-hub** [reply]
Can we have this function as standard in the webui-streamlit version?

**Reply by LIQUIDMIND111** [reply]
> If you want to be able to type in any number you like, then change
> 
> https://github.com/sd-webui/stable-diffusion-webui/blob/e6a9e5d968aad020e54045a0923d8c6aa0b54591/frontend/frontend.py#L39
> 
> to
> 
> `txt2img_batch_count = gr.Number(label='Number of images to generate', precision=0,`
> 
> This will give you a text box to enter any number you like, with a little up/down arrow to progress by one if you prefer clicking.

thanks!!

**Reply by TeaseTest** [reply]
Removing "precision=0" worked for this but it still does not allow the up down arrow click adjustment in decimals, only integers.

This works great, thank you. I would like to replace a lot of the sliders so I'm trying to figure out how to adjust the change increment  with an up/down arrow click. Currently the up or down arrow click adjusts the value but by integer only, I'm looking to figure out how to make it adjust in decimals.

For example, I want a number entry box for CFG scale, with the below code I can type 7.5 in but if I hit the "adjust the number up 1" arrow then it goes to the next integer 8 instead of 7.6.

Working integer number entry for "Number of images to generate".
```
txt2img_batch_count = gr.Number(label='Number of images to generate', precision=0, value=txt2img_defaults['n_iter'])
```

**Reply by TeaseTest** [reply]
I discovered that with this change to enable a typed in integer for the CFG scale, if a decimal number is typed then it is automatically rounded up or down. This is of course very limiting but I found that removing "precision=0" allows for image results with decimal place CFG values.

This solution still does not allow the up down arrow click adjustment in decimals, only integers.

`txt2img_cfg = gr.Number(label='CFG Scale (prompt strength)', value=txt2img_defaults['cfg_scale'], elem_id='cfg_slider')`

precision Optional[int] default: None
Precision to round input/output to. If set to 0, will round to nearest integer and covert type to int. If None, no rounding happens.

**Comment by LIQUIDMIND111** [comment]
Thanks, I'll check it out.
